File: Jumping on the Clouds/jumping-on-the-clouds.py
from typing import List


# Complete the jumpingOnClouds function below.
def jumpingOnClouds(c: List[int]) -> int:
    jumps = 0
    c = c[1:]
    while len(c) > 1:
        part = 2 if c[1] == 0 else 1
        c = c[part:]
        jumps += 1

    return jumps + len(c)


print(jumpingOnClouds([0, 0, 0, 1, 0, 0]))  # 3
print(jumpingOnClouds([0, 0, 1, 0, 0, 1, 0]))  # 4

File: Jumping on the Clouds/jumping-on-the-clouds.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 03.06.19
 * Time: 10:20
 */

/**
 * Determine the minimum number of jumps it to jump from her starting position to the last cloud
 *
 * @param array $c
 * @return int
 */
function jumpingOnClouds(array $c): int
{
    $jumps = 0;
    $c = array_slice($c, 1);
    while (count($c) > 1) {
        $part = $c[1] === 0 ? 2 : 1;
        $c = array_slice($c, $part);
        $jumps++;
    }

    return $jumps + count($c);
}

echo "<pre>jumpingOnClouds([0, 0, 0, 1, 0, 0])\r\n";
echo jumpingOnClouds([0, 0, 0, 1, 0, 0]); // 3
echo "</pre>";

echo "<pre>jumpingOnClouds([0, 0, 1, 0, 0, 1, 0])\r\n";
echo jumpingOnClouds([0, 0, 1, 0, 0, 1, 0]); // 4
echo "</pre>";